app/uploaders/avatar_uploader.rb

```diff
@@ -0,0 +1,91 @@
+# encoding: utf-8
+
+class AvatarUploader < CarrierWave::Uploader::Base
+  if Rails.env.production? || Rails.env.development?
+    storage :fog
+  else
+    storage :file
+  end
+
+  include CarrierWave::MiniMagick
+     process resize_to_fit: [800, 800]
+
+   version :thumb do
+     process resize_to_fill: [200,200]
+   end
+
+   def default_url
+     "/assets/default_images/" + [version_name, "default_photo.jpg"].compact.join('_')
+   end
+
+  uploader = AvatarUploader.new
+
+  uploader.store!(my_file)
+
+  uploader.retrieve_from_store!('my_file.png')
+  u = User.new
+  u.avatar = params[:file] # Assign a file like this, or
+  u.save!
+  u.avatar.url # => '/url/to/file.png'
+  u.avatar.current_path # => 'path/to/file.png'
+  u.avatar.identifier # => 'file.png'
+
+  def extension_white_list
+    %w(jpg jpeg gif png)
+  end
+
+  def default_url(*args)
```

Comment:
Duplicate methods `default_url` at lines `17, 37` detected.<br>Unused method argument - `args`. If it's necessary, use `_` or `_args` as an argument name to indicate that it won't be used. You can also write as `default_url(*)` if you want the method to accept any arguments but don't care about them.